### ISSUE TITLE
[Bug] Cannot instantiate 2 losses with different keys

### ISSUE THREAD

**Issue by matteobettini** [pytorch/rl#1545]
In MARL different agent groups have different loss instances.

It is not currently possible to instantiate the same loss 2 times with different keys

```python
loss_1 = DQNLoss(value_network=nn.Linear(3, 3), action_space="one_hot")
loss_1.set_keys(reward="a")
assert loss_1.tensor_keys.reward == "a"
loss_2 = DQNLoss(value_network=nn.Linear(3, 3), action_space="one_hot")
loss_2.set_keys(reward="b")
assert loss_1.tensor_keys.reward == "a"  # Fails, it is now b
```
